impl/client/src/main/doc-js/pentaho/visual/role/strategies/IMapper.jsdoc

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright 2018 Hitachi Vantara. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The `IMapper` interface describes an object that can map
+ * certain attributes of a certain data set to the value of a visual role,
+ * which is operating in a certain [mode]{@link pentaho.visual.role.strategies.IMapper#mode}.
+ *
+ * @name pentaho.visual.role.strategies.IMapper
+ *
+ * @interface
+ *
+ * @see pentaho.visual.role.strategies.Base
+ */
+
+/**
+ * Gets the input data set.
+ *
+ * @name pentaho.visual.role.strategies.IMapper#inputData
+ * @type {!pentaho.data.ITable}
+ * @readOnly
+ */
+
+/**
+ * Gets the mode of operation of the associated visual role.
+ *
+ * @name pentaho.visual.role.strategies.IMapper#mode
+ * @type {!pentaho.visual.role.Mode}
+ * @readOnly
+ */
+
+/**
+ * Gets a label that describes the visual role as it is mapped.
+ *
+ * @name pentaho.visual.role.strategies.IMapper#label
```

Comment:
I still see this as a function and not as a property of the mapper... But maybe I'm wrong. Is a mapper instance immutable? Will any change in the VR fields or the input data necessarily create a totally new mapper?

Reply:
I didn't review this name or the type of member since we talked about it, in the other day.
The arguments from which a mapper is created are immutable. They do have internal mutable state to be able to satisfy the `invertValue` method. A change to the data or the VR mappings will cause mappers to be invalidated and new mapper instances will be obtained, the next time the `Mapping#mapper` property is read.

So, the label would be immutable for each mapper instance, but `this.model.myRole.mapper.label` can still change over time.

Maybe `expressionLabel`? In the sense that the type of supported mappings corresponds to a combination of the attributes' values that can be written as an expression?